File: src/main/kotlin/dev/murad/shipping/item/EnergyUtil.java
package dev.murad.shipping.item;

import net.minecraft.world.item.ItemStack;
import net.neoforged.neoforge.energy.EnergyStorage;

import java.util.Optional;

import static dev.murad.shipping.setup.ModDataComponents.ENERGY;

public class EnergyUtil {

    private EnergyUtil() {
    }

    public static Optional<EnergyStorage> getEnergyStorage(ItemStack stack) {

        return Optional.ofNullable(stack.getComponents().get(ENERGY.get()));
    }
}

File: src/main/kotlin/dev/murad/shipping/item/EnergyUtil.kt
package dev.murad.shipping.item

import dev.murad.shipping.setup.ModDataComponents
import net.minecraft.world.item.ItemStack
import net.neoforged.neoforge.energy.EnergyStorage
import java.util.*

object EnergyUtil {
    @JvmStatic
    fun getEnergyStorage(stack: ItemStack): Optional<EnergyStorage> {
        return Optional.ofNullable<EnergyStorage>(stack.components.get(ModDataComponents.getEnergyStorage().get()))
    }
}
